test/unit/finance/no_variable_cost_test.rb

```diff
@@ -1,16 +1,51 @@
 require 'test_helper'
 
 class Finance::NoVariableCostClass < ActiveSupport::TestCase
+  attr_reader :contract, :account, :app_plan
 
-  def test_bill_for_variable
-    contract = FactoryBot.build_stubbed(:contract)
-    account  = FactoryBot.build_stubbed(:simple_account)
-    app_plan = FactoryBot.build_stubbed(:application_plan)
+  setup do
+    @contract = FactoryBot.build_stubbed(:contract)
+    @account  = FactoryBot.build_stubbed(:simple_account)
+    @app_plan = FactoryBot.build_stubbed(:application_plan)
+
+    @org_tz = ENV["TZ"]
+  end
+
+  teardown do
+    ENV["TZ"] = @org_tz
+  end
+
+  test "bill for variable" do
+    Timecop.travel(15.days.ago) if Time.now.mday == 1
 
     contract.stubs(:provider_account).returns(account)
     account.stubs(:provider_can_use?).returns(true)
 
     contract.expects(:save).returns(true)
     contract.notify_observers(:bill_variable_for_plan_changed, app_plan)
   end
+
+  test "no variable billing on 1st day of month local time" do
+    # this changes server and client local time to China Standard Time (+8)
+    ENV["TZ"] = "Asia/Shanghai"
+
+    # this is 4 hours before beginning of month in UTC and 4 after local
+    Timecop.travel(Time.zone.now.beginning_of_month - 4.hours)
```

Comment:
Well hopefully there is no DST otherwise it is relatively ±1 hour, otherwise this comment is false. :)
You are comparing UTC (which is absolute) with timezone which are affected by daylight saving.

Reply:
I chose such a time difference so that DTS will not affect the outcome. Also IIRC China timezone does not use DTS at all.

Reply:
It is intentional - to make sure that billing uses local time and not UTC. If we used UTC in billing, then test would fail.

Reply:
https://www.timeanddate.com/time/change/china/beijing

DST is no longer in use.

Reply:
Comment should be always correct :)